index.js

```diff
@@ -46,4 +42,8 @@ Object.defineProperty(Layer.prototype, 'handle', {
   },
 });
 
-patchRouterParam();
+const originalParam = Router.prototype.constructor.param;
+Router.prototype.constructor.param = function param(name, fn) {
+  fn = wrap(fn);
+  return originalParam.call(this, name, fn);
```

Comment:
Do we not change the value of `this` by removing this section from the `wrap` closure?

Reply:
Value of `this` in this case is determined by `Router#param` callsite and explicitly forwarded to original implementation using `.call` binding so it is preserved. Moving it out of a function doesn't change it.